cmd/root.go

```diff
@@ -41,11 +44,20 @@ func initRootConfig() {
 
 // parseConfig reads the flags and ENV vars
 func parseConfig(cmd *cobra.Command, args []string) {
+
+	err := k.Load(env.Provider("", ".", func(s string) string {
+		// First replace the double underscore with . as the hierarchy delimiter
+		s = strings.Replace(strings.ToLower(s), "__", ".", -1)
+		// Now replace the remaining single underscore with dashes to allow flags with dashes in it.
+		return strings.Replace(s, "_", "-", -1)
```

Comment:
This code is hard to understand and the comments help little. How about something like:
```suggestion
		// environment variables can contain hierarchies and <???>,
		// which are read from the environment variable values, separated by
		// booo ...
		// Example: foo.bar --baz  ➜   foo__bar_baz
		s = strings.Replace(strings.ToLower(s), "__", ".", -1).Replace(s, "_", "-", -1)
		return s
```

Reply:
Agree on the commenting intention.

However, a one liner is not possible as suggested, since `Replace` is a function from the `strings` package, which returns a new string. Strings do not have the `Replace` method itself.

Reply:
True. :roll_eyes: So, fine with me then.